src/Simulation/Simulators/QCTraceSimulator/QCTraceSimulator.cs

```diff
@@ -98,6 +98,14 @@ public class QCTraceSimulatorConfiguration
         /// </summary>
         public bool UseWidthCounter = false;
 
+        /// <summary>
+        /// Constrols if depth or width optimization is favored.
+        /// If set to <c>true</c>, resulting circuit is optimized for depth by discouraging qubit reuse.
+        /// If set to <c>false</c>, resulting circuit is optimized for width be encouraging qubit reuse.
+        /// Optimization is only limited to reuse of qubits after they are released by user code.
+        /// </summary>
+        public bool OptimizeDepth;
```

Comment:
Good question. This translates directly to QubitManager option EncourageReuse so I decided to use just bool. At first I wanted to name it just EncourageQubitReuse, but it would have been a different layer of abstraction. So I named it OptimizeDepth as this was the major driver for this feature. Now that you pointed this out, this level of abstraction may have its own value and makes sense.

Reply:
On the closer consideration - it will complicate code. This value is passed through layers (QCTraceSimulatorConfiguration, QCTraceSimulatorCoreConfiguration) and user of the top layer may not need/want to know about low layer. So either two emums need to be introduced and converted or the user of the top layer will have to know about the low layer. Existing bool is simple and reasonable enough.